src/plugins/dashboard/public/dashboard_api/unified_search_manager.ts

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  COMPARE_ALL_OPTIONS,
+  Filter,
+  Query,
+  TimeRange,
+  compareFilters,
+  isFilterPinned,
+} from '@kbn/es-query';
+import {
+  BehaviorSubject,
+  Observable,
+  Subject,
+  Subscription,
+  combineLatest,
+  debounceTime,
+  distinctUntilChanged,
+  finalize,
+  map,
+  of,
+  switchMap,
+  tap,
+} from 'rxjs';
+import fastIsEqual from 'fast-deep-equal';
+import { PublishingSubject, StateComparators } from '@kbn/presentation-publishing';
+import { ControlGroupApi } from '@kbn/controls-plugin/public';
+import { cloneDeep } from 'lodash';
+import {
+  GlobalQueryStateFromUrl,
+  RefreshInterval,
+  connectToQueryState,
+  syncGlobalQueryStateWithUrl,
+} from '@kbn/data-plugin/public';
+import { cleanFiltersForSerialize } from '@kbn/presentation-util-plugin/public';
+import moment, { Moment } from 'moment';
+import { dataService } from '../services/kibana_services';
+import { DashboardCreationOptions, DashboardState } from './types';
+import { DEFAULT_DASHBOARD_INPUT, GLOBAL_STATE_STORAGE_KEY } from '../dashboard_constants';
+
+export function initializeUnifiedSearchManager(
+  initialState: DashboardState,
+  controlGroupApi$: PublishingSubject<ControlGroupApi | undefined>,
+  waitForPanelsToLoad$: Observable<void>,
+  getLastSavedState: () => DashboardState | undefined,
+  creationOptions?: DashboardCreationOptions
+) {
+  const {
+    queryString,
+    filterManager,
+    timefilter: { timefilter: timefilterService },
+  } = dataService.query;
+
+  const controlGroupReload$ = new Subject<void>();
+  const filters$ = new BehaviorSubject<Filter[] | undefined>(undefined);
+  const panelsReload$ = new Subject<void>();
+  const query$ = new BehaviorSubject<Query | undefined>(initialState.query);
+  // setAndSyncQuery method not needed since query synced with 2-way data binding
+  function setQuery(query: Query) {
+    if (!fastIsEqual(query, query$.value)) {
+      query$.next(query);
+    }
+  }
+  const refreshInterval$ = new BehaviorSubject<RefreshInterval | undefined>(
+    initialState.refreshInterval
+  );
+  function setRefreshInterval(refreshInterval: RefreshInterval) {
+    if (!fastIsEqual(refreshInterval, refreshInterval$.value)) {
+      refreshInterval$.next(refreshInterval);
+    }
+  }
+  function setAndSyncRefreshInterval(refreshInterval: RefreshInterval | undefined) {
```

Comment:
`timefilterService` is not always available. For example, a portable dashboard may specify `useUnifiedSearchIntegration: false`. In these cases, `refreshInterval$` would not get updated if we soloy rely on the value flowing through the `timefilterService.getRefreshIntervalUpdate$()` subscription. This method exists to considerate these places into a single method to avoid code duplication.